src/sentry/web/frontend/base.py

```diff
@@ -41,83 +35,100 @@
 
 
 class OrganizationMixin:
+    # This attribute will only be set once determine_active_organization is called.  Subclasses should likely invoke
+    # that method, passing along the organization_slug context that might exist (or might not).
+    active_organization: Optional[ApiOrganization]
+
     # TODO(dcramer): move the implicit organization logic into its own class
     # as it's only used in a single location and over complicates the rest of
     # the code
-    def get_active_organization(self, request: Request, organization_slug=None):
+    def determine_active_organization(self, request: Request, organization_slug=None) -> NoReturn:
```

Comment:
@corps Now that `determine_active_organization()` is refactored. I'm wondering if it's worthwhile to write tests surrounding this function?

For example, I'm interested to see if the customer domain (i.e. `request.subdomain`) is respected properly, and not overridden. I'm willing to write these kinds of tests.

Reply:
Yeah, I'm willing to write some tests for it now in my branch.  Actually, after testing this locally and playing a debugger I am starting to better understand all its various cases.  Let me know if there's any cases you'd like me to cover, I'll start adding more tests.

Reply:
It does seem that request.subdomain is respected currently in the auth workfows, there is some coverage for that.